src/battle_ai_switch_items.c

```diff
@@ -240,21 +240,21 @@ static bool8 FindMonThatAbsorbsOpponentsMove(void)
             if (absorbingTypeAbilities[j] == monAbility && Random() & 1)
             {
                 // we found a mon.
-                *(gBattleStruct->AI_monToSwitchIntoId + gActiveBattler) = i;
```

Comment:
not necessarily in this PR, but we should change `*(gBattleStruct->AI_monToSwitchIntoId + battler);` to `gBattleStruct->AI_monToSwitchIntoId[battler];` and other similar lines

Reply:
Yeah, I agree, these should be changed.